### PR TITLE
Add Whisper support in BeamSearch Op

### PR DESCRIPTION
**Pull request by petermcaughan**
### Description
Adjust various code paths to allow Whisper model to function with BeamSearch op.

Approach: Add a new kModelType enum value in IGenerationParameters as so:
#### New: 0 = GPT2, 1 = T5, 2 = Whisper

When the user assigns this attribute value to 2, various shape and type checks are changed to accommodate Whisper inputs.

### Motivation and Context
BeamSearch is currently designed to function with BERT-based models with inputs as vocab tokens, and needs changes to function with Whisper inputs (3-D float values processed from audio data).

### CODE REVIEW

**Comment by tianleiwu** [comment]
Please add pull request for exporter script first. You can follow the example of T5:
https://github.com/microsoft/onnxruntime/tree/main/onnxruntime/python/tools/transformers/models/t5

Try to use similar inputs and outputs of T5.

File: onnxruntime/contrib_ops/cpu/transformers/beam_search.cc [hunk 1]
```diff
@@ -148,6 +150,37 @@ Status BeamSearch::SetupSubgraphExecutionInfo(const SessionState& session_state,
                                         t5_decoder_subgraph_->num_layers);
     }
   }
+  else if (parameters_.model_type == IGenerationParameters::kModelTypeWhisper) {
+    if (attribute_name == "encoder") {
+      ORT_ENFORCE(t5_encoder_subgraph_ == nullptr,
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cpu/transformers/beam_search.cc:155]: >t5_encoder_subgraph_ [](http://example.com/codeflow?start=18&length=20)

Why do we need a specific type for Whisper? Aren't T5 and Whisper the same type?

**Review by yufenglee — DISMISSED** [review]

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 2]
```diff
@@ -410,8 +411,19 @@ void BeamSearchShapeInference(ONNX_NAMESPACE::InferenceContext& ctx) {
   }
   auto& input_ids_shape = getInputShape(ctx, 0);
   auto& input_ids_dims = input_ids_shape.dim();
-  if (input_ids_dims.size() != 2) {
-    fail_shape_inference("Inputs 0 shall be 2 dimensions");
+  auto model_type_attr = ctx.getAttribute("model_type");
+  int64_t model_type = model_type_attr ? static_cast<int64_t>(model_type_attr->i()) : -1;
+  if (model_type ==  onnxruntime::contrib::transformers::IGenerationParameters::kModelTypeWhisper) {
+     if (input_ids_dims.size() != 3)
```
Comment (tianleiwu) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc:417]: format the code

**Pull request closed without merging**